src/utils/trpc.ts

```diff
@@ -56,11 +56,12 @@ export const custom401Link: TRPCLink<AppRouter> = () => {
   }
 }
 
-const handleErrorsOnClient = (error: unknown): boolean => {
-  if (typeof window === 'undefined') return false
-  if (!(error instanceof TRPCClientError)) return false
+const isErrorRetryableOnClient = (error: unknown): boolean => {
+  if (typeof window === 'undefined') return true
+  if (!(error instanceof TRPCClientError)) return true
   const res = TRPCWithErrorCodeSchema.safeParse(error)
-  return res.success && NON_RETRYABLE_ERROR_CODES.has(res.data)
+  if (res.success && NON_RETRYABLE_ERROR_CODES.has(res.data)) return false
+  return true
```

Comment:
nit: the mathematical negation of `a && b` is `!a || !b`, if we want to make it less confusing. hurray, discrete algebra...

```suggestion
  return(!res.success || !NON_RETRYABLE_ERROR_CODES.has(res.data))
```

Reply:
haha the way my brain parse it is:
- If `safeParse` is successful and the result is one of the non-retryable error codes, don't retry
- for everything else, just retry

but happy to defer to the reviewer!